pkg/xds/generator/outbound_proxy_generator.go

```diff
@@ -62,8 +64,12 @@ func (g OutboundProxyGenerator) Generate(ctx context.Context, _ *model.ResourceS
 
 		protocol := inferProtocol(xdsCtx.Mesh, clusters)
 		switch protocol {
-		case core_mesh.ProtocolHTTP, core_mesh.ProtocolHTTP2:
-			if hasMeshRoutes {
+		case core_mesh.ProtocolHTTP, core_mesh.ProtocolHTTP2, core_mesh.ProtocolGRPC:
+			if hasMeshHTTPRoutes {
+				continue
+			}
+		case core_mesh.ProtocolUnknown, core_mesh.ProtocolTCP:
```

Comment:
I think kafka missing
```suggestion
		case core_mesh.ProtocolUnknown, core_mesh.ProtocolKafka, core_mesh.ProtocolTCP:
```

Reply:
I think we have a code change in MeshTCPRoute(and test) which allows kafka protocol https://github.com/kumahq/kuma/blob/2df58666a6a80e39b78e7ac8a0f2b9c59f8e244b/pkg/plugins/policies/meshtcproute/plugin/v1alpha1/testdata/kafka.listeners.golden.yaml